### PR TITLE
DB dump tool

### PR DESCRIPTION
**Pull request by shans**

### CODE REVIEW

File: src/tools/db-dump/db-dump-data-model.ts [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+ * http://polymer.github.io/PATENTS.txt
+ */
+import Database from 'better-sqlite3';
+import {exit} from 'process';
+import {assert} from 'console';
+
+
+interface Dictionary<T> {
+  [key: string]: T;
+}
```
Comment (galganif) [inline src/tools/db-dump/db-dump-data-model.ts:17]: you could use the one from hot.ts. maybe add the number one there as well

File: src/tools/db-dump/db-dump-data-model.ts [hunk 2]
```diff
@@ -0,0 +1,484 @@
...
+
+interface NumberDictionary<T> {
+  [key: number]: T;
+}
+
+type Entity = {
+  storage_key_id: number;
+  entity_id: string;
+
+  clean_entity_id: string;
```
Comment (galganif) [inline src/tools/db-dump/db-dump-data-model.ts:27]: a comment on the difference between clean and not?

File: src/tools/db-dump/db-dump-data-model.ts [hunk 3]
```diff
@@ -0,0 +1,484 @@
...
+  clean_storage_key: string;
+}
+
+type Type = {
+  id: number,
+  name: string,
+  is_primitive: number,
+
+  fields?: Field[],
+  subFields?: Field[]
```
Comment (galganif) [inline src/tools/db-dump/db-dump-data-model.ts:46]: what's a subfield?

File: src/tools/db-dump/db-dump-data-model.ts [hunk 4]
```diff
@@ -0,0 +1,484 @@
...
+        break;
+    }
+    console.log(`${indent}${field.name}: ${name}`);
+    field.subFields && field.subFields.forEach(field => this.printField(field, indent + '  '));
+  }
+
+  // b/170674301 Sometimes primitive types get removed from the DB. Check, warn, and synthesize.
+  private adjustTypes() {
+    if (!this.types[0]) {
+      console.warn(`primitive type records missing! See b/170674301 for more information.`);
```
Comment (galganif) [inline src/tools/db-dump/db-dump-data-model.ts:298]: we removed primitive types from the db entirely, you can remove this warning

**Review by galganif — APPROVED** [review]

**Comment by shans** [comment]
@SHeimlich Filippo suggested adding you as a reviewer to help assess usefulness of this dump tool for devTools in general.

**Review by SHeimlich — APPROVED** [review]
CC: @piotrswigon

**Pull request closed without merging**